modules/electromagnetics/src/kernels/FuncDiffusion.C

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FuncDiffusion.h"
+#include "Function.h"
+
+registerMooseObject("ElectromagneticsApp", FuncDiffusion);
+
+InputParameters
+FuncDiffusion::validParams()
+{
+  InputParameters params = Diffusion::validParams();
+  params.addClassDescription("The Laplacian operator with a function coefficient.");
+  params.addParam<FunctionName>("func", 1.0, "Function multiplier for diffusion term.");
+  return params;
+}
+
+FuncDiffusion::FuncDiffusion(const InputParameters & parameters)
+  : Diffusion(parameters),
+
+    _func(getFunction("func"))
+
+{
+}
+
+Real
+FuncDiffusion::computeQpResidual()
+{
+  return _func.value(_t, _q_point[_qp]) * Diffusion::computeQpResidual();
```

Comment:
This could be heresy, but I think we should just add the function parameter to the framework `Diffusion` object as well. I know there has been a historical argument for keeping the framework objects as simple as possible, but that argument doesn't resonate with me. Overly simplistic objects should be test objects. Usable (these *can* be simple still) objects that everyone can benefit from are great candidates for framework objects as opposed to test objects. If you asked my opinion, I would say that the current state of `Diffusion` should be a test object instead of a framework object. No one uses `Diffusion` in any real physics input.

Reply:
I've gone back and forth, because I share your sentiment on objects like this. It seems like we have all these special case objects that could just be combined. I also planned to move this into the framework on the next revision, so perhaps I should just combine this with `Diffusion` proper.

Reply:
I can buy that, but I also think adding a function coefficient to it isn't going to destroy how simple it is. I can leave it alone and just move this into framework with a name change.

Reply:
> diffusion needs to stay as simple as possible, this is the number one kernel we can show to people

It's poorly named. No one does diffusion with a unity coefficient. If we wanted to keep this the same forever, we should have named it something like `Laplace`

Reply:
I moved this into framework and changed the name to `FunctionDiffusion` in a pending commit.